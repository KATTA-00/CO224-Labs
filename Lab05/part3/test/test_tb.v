
`include "test.v"

module test_tb;

    reg RESET, CLK;
    reg [31:0] INSTRUCTION;
    wire [31:0] PC;

    reg [7:0] DATA;
    wire [7:0] OUT;


    cpu CPU(PC, INSTRUCTION, CLK, RESET);


    mux m(8'd1, 8'd2, 1'd0, OUT);

    always
    #5 CLK = ~CLK;

    initial
    #30 $finish;

    initial begin
        CLK = 0;
        $monitor($time, " %b %b %b %b %b %b", CLK, PC, INSTRUCTION, RESET, DATA, OUT);

        $dumpfile("test.vcd");
		$dumpvars(0, test_tb);

        #8 DATA = 2;

    end



endmodule

module two_comp(DATA, OUT);

    input [7:0] DATA;
    output [7:0] OUT;

    assign OUT = ~DATA + 1;

endmodule

module mux(DATA1, DATA2, SELECT, OUTPUT);

    input [7:0] DATA1, DATA2;
    input SELECT;
    output reg [7:0] OUTPUT;

    always @(SELECT) begin 

        if (SELECT) 
            OUTPUT = DATA1;
        else
            OUTPUT = DATA2;

    end

endmodule