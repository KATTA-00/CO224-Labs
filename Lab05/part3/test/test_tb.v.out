#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "F:\iverilog\lib\ivl\system.vpi";
:vpi_module "F:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "F:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "F:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "F:\iverilog\lib\ivl\va_math.vpi";
S_000002808da89fd0 .scope module, "test_tb" "test_tb" 2 4;
 .timescale 0 0;
v000002808dadf9e0_0 .var "CLK", 0 0;
v000002808dae05c0_0 .var "DATA", 7 0;
v000002808dadfa80_0 .var "INSTRUCTION", 31 0;
v000002808dae0340_0 .net "OUT", 7 0, v000002808dae0160_0;  1 drivers
v000002808dadfda0_0 .net "PC", 31 0, v000002808da8a390_0;  1 drivers
v000002808dae0660_0 .var "RESET", 0 0;
S_000002808dbbe530 .scope module, "CPU" "cpu" 2 14, 3 2 0, S_000002808da89fd0;
 .timescale 0 0;
    .port_info 0 /OUTPUT 32 "PC";
    .port_info 1 /INPUT 32 "INSTRUCTION";
    .port_info 2 /INPUT 1 "CLK";
    .port_info 3 /INPUT 1 "RESET";
v000002808dae07a0_0 .net "CLK", 0 0, v000002808dadf9e0_0;  1 drivers
v000002808dae0480_0 .net "INSTRUCTION", 31 0, v000002808dadfa80_0;  1 drivers
v000002808dae0700_0 .net "PC", 31 0, v000002808da8a390_0;  alias, 1 drivers
v000002808dae02a0_0 .net "RESET", 0 0, v000002808dae0660_0;  1 drivers
S_000002808dbbe6c0 .scope module, "p" "pc" 3 9, 3 19 0, S_000002808dbbe530;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "RESET";
    .port_info 1 /INPUT 1 "CLK";
    .port_info 2 /OUTPUT 32 "PC";
v000002808da8a2f0_0 .net "CLK", 0 0, v000002808dadf9e0_0;  alias, 1 drivers
v000002808da8a390_0 .var "PC", 31 0;
v000002808dbbd7c0_0 .var "PC_NEXT", 31 0;
v000002808dbbd860_0 .net "RESET", 0 0, v000002808dae0660_0;  alias, 1 drivers
v000002808dbbd900_0 .net "adder_out", 31 0, L_000002808dae0520;  1 drivers
E_000002808da8c860 .event posedge, v000002808da8a2f0_0;
E_000002808da8d4a0 .event anyedge, v000002808dbbe8f0_0;
S_000002808dbbd630 .scope module, "pc_adder" "pc_add" 3 27, 3 41 0, S_000002808dbbe6c0;
 .timescale 0 0;
    .port_info 0 /INPUT 32 "PC";
    .port_info 1 /OUTPUT 32 "adder_out";
v000002808da794e0_0 .net "PC", 31 0, v000002808da8a390_0;  alias, 1 drivers
L_000002808dae0868 .functor BUFT 1, C4<00000000000000000000000000000001>, C4<0>, C4<0>, C4<0>;
v000002808dbbe850_0 .net/2u *"_ivl_0", 31 0, L_000002808dae0868;  1 drivers
v000002808dbbe8f0_0 .net "adder_out", 31 0, L_000002808dae0520;  alias, 1 drivers
L_000002808dae0520 .delay 32 (1,1,1) L_000002808dae0520/d;
L_000002808dae0520/d .arith/sum 32, v000002808da8a390_0, L_000002808dae0868;
S_000002808da8c190 .scope module, "m" "mux" 2 17, 2 49 0, S_000002808da89fd0;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "DATA1";
    .port_info 1 /INPUT 8 "DATA2";
    .port_info 2 /INPUT 1 "SELECT";
    .port_info 3 /OUTPUT 8 "OUTPUT";
L_000002808dae08b0 .functor BUFT 1, C4<00000001>, C4<0>, C4<0>, C4<0>;
v000002808dae03e0_0 .net "DATA1", 7 0, L_000002808dae08b0;  1 drivers
L_000002808dae08f8 .functor BUFT 1, C4<00000010>, C4<0>, C4<0>, C4<0>;
v000002808dadf8a0_0 .net "DATA2", 7 0, L_000002808dae08f8;  1 drivers
v000002808dae0160_0 .var "OUTPUT", 7 0;
L_000002808dae0940 .functor BUFT 1, C4<0>, C4<0>, C4<0>, C4<0>;
v000002808dadf940_0 .net "SELECT", 0 0, L_000002808dae0940;  1 drivers
E_000002808da8d0e0 .event anyedge, v000002808dadf940_0;
S_000002808da8a160 .scope module, "two_comp" "two_comp" 2 40;
 .timescale 0 0;
    .port_info 0 /INPUT 8 "DATA";
    .port_info 1 /OUTPUT 8 "OUT";
o000002808da8f488 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
L_000002808da77790 .functor NOT 8, o000002808da8f488, C4<00000000>, C4<00000000>, C4<00000000>;
v000002808dadfb20_0 .net "DATA", 7 0, o000002808da8f488;  0 drivers
v000002808dadfbc0_0 .net "OUT", 7 0, L_000002808dadfe40;  1 drivers
v000002808dadfd00_0 .net *"_ivl_0", 7 0, L_000002808da77790;  1 drivers
L_000002808dae0988 .functor BUFT 1, C4<00000001>, C4<0>, C4<0>, C4<0>;
v000002808dadfc60_0 .net/2u *"_ivl_2", 7 0, L_000002808dae0988;  1 drivers
L_000002808dadfe40 .arith/sum 8, L_000002808da77790, L_000002808dae0988;
    .scope S_000002808dbbe6c0;
T_0 ;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v000002808da8a390_0, 0, 32;
    %end;
    .thread T_0;
    .scope S_000002808dbbe6c0;
T_1 ;
    %wait E_000002808da8d4a0;
    %load/vec4 v000002808dbbd900_0;
    %store/vec4 v000002808dbbd7c0_0, 0, 32;
    %jmp T_1;
    .thread T_1, $push;
    .scope S_000002808dbbe6c0;
T_2 ;
    %wait E_000002808da8c860;
    %delay 1, 0;
    %load/vec4 v000002808dbbd7c0_0;
    %store/vec4 v000002808da8a390_0, 0, 32;
    %jmp T_2;
    .thread T_2;
    .scope S_000002808da8c190;
T_3 ;
    %wait E_000002808da8d0e0;
    %load/vec4 v000002808dadf940_0;
    %flag_set/vec4 8;
    %jmp/0xz  T_3.0, 8;
    %load/vec4 v000002808dae03e0_0;
    %store/vec4 v000002808dae0160_0, 0, 8;
    %jmp T_3.1;
T_3.0 ;
    %load/vec4 v000002808dadf8a0_0;
    %store/vec4 v000002808dae0160_0, 0, 8;
T_3.1 ;
    %jmp T_3;
    .thread T_3, $push;
    .scope S_000002808da89fd0;
T_4 ;
    %delay 5, 0;
    %load/vec4 v000002808dadf9e0_0;
    %inv;
    %store/vec4 v000002808dadf9e0_0, 0, 1;
    %jmp T_4;
    .thread T_4;
    .scope S_000002808da89fd0;
T_5 ;
    %delay 30, 0;
    %vpi_call 2 23 "$finish" {0 0 0};
    %end;
    .thread T_5;
    .scope S_000002808da89fd0;
T_6 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000002808dadf9e0_0, 0, 1;
    %vpi_call 2 27 "$monitor", $time, " %b %b %b %b %b %b", v000002808dadf9e0_0, v000002808dadfda0_0, v000002808dadfa80_0, v000002808dae0660_0, v000002808dae05c0_0, v000002808dae0340_0 {0 0 0};
    %vpi_call 2 29 "$dumpfile", "test.vcd" {0 0 0};
    %vpi_call 2 30 "$dumpvars", 32'sb00000000000000000000000000000000, S_000002808da89fd0 {0 0 0};
    %delay 8, 0;
    %pushi/vec4 2, 0, 8;
    %store/vec4 v000002808dae05c0_0, 0, 8;
    %end;
    .thread T_6;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "test_tb.v";
    "./test.v";
